test/functional/test_framework/test_node.py

```diff
@@ -267,6 +283,29 @@ def assert_debug_log(self, expected_msgs):
                 if re.search(re.escape(expected_msg), log, flags=re.MULTILINE) is None:
                     self._raise_assertion_error('Expected message "{}" does not partially match log:\n\n{}\n\n'.format(expected_msg, print_log))
 
+    @contextlib.contextmanager
+    def assert_memory_usage_stable(self, perc_increase_allowed=0.03):
+        """Context manager that allows the user to assert that a node's memory usage (RSS)
+        hasn't increased beyond some threshold percentage.
+        """
+        before_memory_usage = self.get_mem_rss()
+
+        yield
+
+        after_memory_usage = self.get_mem_rss()
+
+        if not (before_memory_usage and after_memory_usage):
+            self.log.warning("Unable to detect memory usage (RSS) - skipping memory check.")
```

Comment:
If this is expected to happen sometimes, (I would assume because `ps` isn't supported), might want to leave a comment as to why this check is being skipped rather than throwing an exception.

Reply:
No exception thrown here - just a warning log statement and early return.

Reply:
Sorry - I didn't phrase that right. I meant, it may make sense to leave a comment as to why its okay to skip the check, and not just throw an exception (as some might expect to happen).

Reply:
Actually with the log from the CalledProcessError its fine, so please disregard :)